mosaicml/streaming
# Add Build publish to PyPI and create GitHub release workflow

## Description
- Add Build and publish to PyPI workflow
- Add create GitHub release workflow
- Update existing actions to latest version

## Usage
The `pypi-release.yaml` workflow will be kick in with the below commands
```
git tag -a "v0.0.2b" -m "pypi package versioning"
git push --tags
```

With the above 2 lines, the GitHub actions will do the following
- Run pre-commit for linting issue
- Run pytest to ensure it passes
- Create a pypi wheel, check package description on PyPI using `twine`, uploads the pypi wheel to pypi website.
- Create a GitHub Release
